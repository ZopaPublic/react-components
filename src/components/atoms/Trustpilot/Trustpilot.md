### Summary

Use this to render Trustpilot logo with stars.

### Examples

- It defaults to 5 stars...

```tsx
import { Trustpilot } from '@zopauk/react-components';

<Trustpilot />;
```

- ...but you can decide the number as well.

```tsx
import { useState, useEffect } from 'react';
import { Trustpilot } from '@zopauk/react-components';

const RatingsRoulette = () => {
  const [rating, setRating] = useState(0);

  const shuffleRating = () => {
    setRating((r) => (r === 5 ? 0 : r === 0 ? r + 1 : r + 0.5));
  };

  useEffect(() => {
    const interval = setInterval(shuffleRating, 750);
    return () => {
      clearInterval(interval);
    };
  }, []);

  return <Trustpilot rating={rating} />;
};

<RatingsRoulette />;
```
