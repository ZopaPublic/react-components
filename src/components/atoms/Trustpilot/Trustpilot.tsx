import React from 'react';
import Logo from './Logo/Logo';
import Stars, { TrustPilotStarsProps } from './Stars/Stars';

/**
 * @visibleName Trustpilot
 */
const Trustpilot = ({ rating, className }: TrustPilotStarsProps) => (
  <div className={className}>
    <Logo className="mr-1" />
    <Stars className="ml-1" rating={rating} />
  </div>
);

export default Object.assign(Trustpilot, {
  Logo,
  Stars,
});
