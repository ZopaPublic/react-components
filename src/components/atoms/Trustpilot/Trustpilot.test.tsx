import axe from '../../../../axe-helper';
import React from 'react';
import { render } from '@testing-library/react';
import Trustpilot from './Trustpilot';
import 'jest-styled-components';

describe('<Trustpilot />', () => {
  it('renders without a11y violations', async () => {
    const { container } = render(<Trustpilot />);
    const results = await axe(container.innerHTML);
    expect(results).toHaveNoViolations();
    expect(container).toMatchSnapshot();
  });
});
