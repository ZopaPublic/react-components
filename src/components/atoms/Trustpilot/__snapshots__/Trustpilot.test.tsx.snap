// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Trustpilot /> renders without a11y violations 1`] = `
<div>
  <div>
    <svg
      aria-labelledby="trustpilot-logo-title trustpilot-logo-desc"
      class="mr-1"
      height="16px"
      viewBox="0 0 212 53"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title
        id="trustpilot-logo-title"
      >
        Trustpilot Logo
      </title>
      <desc
        id="trustpilot-logo-desc"
      >
        An illustration of the Trustpilot Logo
      </desc>
      <path
        d="M55.7 18.7h21.5v4.1h-8.4v22.9h-4.6V22.8h-8.4zm20.5 7.5h4V30h.1c.1-.6.4-1.1.7-1.5l1.3-1.4c.5-.4 1.1-.8 1.7-1s1.2-.4 1.9-.4h1l.6.1v4.1l-.9-.1-.9-.1c-.7 0-1.4.2-2 .4s-1.2.7-1.6 1.3c-.5.6-.8 1.3-1.1 2.1s-.4 1.8-.4 2.9v9.3h-4.2zm30.7 19.4h-4.2v-2.7h-.1c-.5 1-1.3 1.8-2.3 2.4s-2.1.9-3.1.9c-2.5 0-4.4-.6-5.5-1.9S90 41.1 90 38.5V26.2h4.2v11.9c0 1.7.3 2.9 1 3.6.6.7 1.6 1.1 2.7 1.1.9 0 1.6-.1 2.2-.4s1.1-.6 1.4-1.1c.4-.5.6-1 .8-1.7.2-.6.2-1.3.2-2.1V26.2h4.2v19.4zm7.2-6.2c.1 1.3.6 2.1 1.4 2.6s1.8.8 2.9.8l1.3-.1 1.4-.4c.4-.2.8-.4 1.1-.8.3-.3.4-.8.4-1.3s-.2-1-.6-1.3c-.4-.4-.8-.6-1.4-.9l-2-.6-2.3-.5-2.3-.6c-.7-.2-1.4-.6-2-1s-1.1-1-1.4-1.6c-.4-.7-.5-1.5-.5-2.5 0-1.1.3-1.9.8-2.7.5-.7 1.2-1.3 1.9-1.7a12.4 12.4 0 018.1-.9c.9.2 1.7.6 2.4 1a6.2 6.2 0 012.7 4.7h-4c-.2-1-.7-1.7-1.4-2.1s-1.6-.5-2.5-.5l-1.1.1-1.2.3c-.3.1-.7.3-.9.6s-.4.6-.4 1c0 .5.2 1 .5 1.3.4.3.8.6 1.4.8l2 .5 2.3.5 2.3.6c.7.2 1.4.6 2 1s1 1 1.4 1.6.5 1.5.5 2.4c0 1.2-.3 2.1-.8 2.9s-1.2 1.5-2 2c-.9.5-1.8.9-2.8 1.1a15.3 15.3 0 01-6.4-.1c-1-.3-1.9-.7-2.7-1.3-.7-.6-1.4-1.3-1.8-2.1s-.7-1.8-.7-3zm14-13.2h3.2v-5.9h4.2v5.9h3.8v3.2h-3.8v10.4l.1 1.2c0 .3.1.6.3.8.1.2.3.4.6.5l1.1.2h.9l.9-.1v3.3l-1.4.2-1.4.1c-1.1 0-2-.1-2.7-.3s-1.2-.5-1.6-1c-.4-.4-.6-.9-.8-1.6a15 15 0 01-.2-2.1V29.4h-3.2zm14.2 0h4v2.6h.1c.6-1.1 1.4-1.9 2.5-2.4s2.2-.7 3.5-.7c1.5 0 2.8.3 3.9.8s2.1 1.3 2.8 2.2c.7 1 1.3 2.1 1.7 3.3a13.4 13.4 0 01.1 7.8c-.3 1.2-.8 2.3-1.5 3.3s-1.5 1.7-2.6 2.3-2.3.9-3.7.9l-1.9-.2-1.8-.6c-.6-.2-1.1-.6-1.6-1s-.9-.9-1.2-1.4h-.1v9.7h-4.2zm14.8 9.7c0-.9-.1-1.7-.3-2.6-.2-.8-.5-1.5-1-2.2-.4-.6-1-1.2-1.7-1.5-.7-.4-1.5-.6-2.3-.6-1.8 0-3.1.6-4 1.9s-1.3 2.9-1.3 5c0 1 .1 1.9.4 2.7s.6 1.6 1.1 2.2a4 4 0 001.7 1.4c.7.4 1.4.5 2.3.5 1 0 1.8-.2 2.5-.6s1.2-.9 1.6-1.6c.4-.6.7-1.4.9-2.2l.1-2.4zm7.5-17.2h4.2v4.1h-4.2zm0 7.5h4.2v19.5h-4.2V26.2zm8-7.5h4.2v26.9h-4.2zm17.2 27.5c-1.5 0-2.9-.3-4.1-.8s-2.2-1.2-3.1-2.1c-.8-.9-1.5-2-1.9-3.2-.4-1.3-.7-2.6-.7-4.1a11 11 0 012.6-7.3c.8-.9 1.9-1.6 3.1-2.1s2.6-.8 4.1-.8 2.9.3 4.1.8 2.2 1.2 3.1 2.1c.8.9 1.5 2 1.9 3.2.4 1.3.7 2.6.7 4.1s-.2 2.9-.7 4.1a8.5 8.5 0 01-5 5.3c-1.2.5-2.6.8-4.1.8zm0-3.4c.9 0 1.8-.2 2.4-.6.7-.4 1.3-.9 1.7-1.6.4-.6.8-1.4 1-2.2a10.7 10.7 0 000-5 6.2 6.2 0 00-2.7-3.8c-.7-.4-1.5-.6-2.5-.6-.9 0-1.8.2-2.5.6s-1.3.9-1.7 1.6c-.4.6-.8 1.4-1 2.2a10 10 0 000 5c.2.8.5 1.6 1 2.2.4.6 1 1.2 1.7 1.6.8.4 1.7.6 2.6.6zm10.9-16.6h3.2v-5.9h4.2v5.9h3.8v3.2h-3.8v10.4l.1 1.2c0 .3.1.6.3.8.1.2.3.4.6.5l1.1.2h.9l.9-.1v3.3l-1.4.2-1.4.1c-1.1 0-2-.1-2.7-.3s-1.2-.5-1.6-1c-.4-.4-.6-.9-.8-1.6a15 15 0 01-.2-2.1V29.4h-3.2z"
      />
      <path
        d="M50.8 18.7H31.4L25.4 0l-6 18.7H0l15.7 11.6-6 18.7 15.7-11.6L41.1 49l-6-18.7z"
        fill="#00B679"
      />
      <path
        d="M36.4 34.5l-1.3-4.2-9.7 7.1z"
        fill="#02917c"
      />
    </svg>
    <svg
      aria-labelledby="trust-pilot-rating-title trust-pilot-rating-desc"
      class="ml-1"
      height="20px"
      role="img"
      viewBox="0 0 512 96"
      xmlns="http://www.w3.org/2000/svg"
      xmlns:xlink="http://www.w3.org/1999/xlink"
    >
      <title
        id="trust-pilot-rating-title"
      >
        Trustpilot Five stars rating
      </title>
      <desc
        id="trust-pilot-rating-desc"
      >
        An illustration of the Trustpilot Stars Logo
      </desc>
      <g
        fill="none"
        fill-rule="evenodd"
        id="Trustpilot_ratings_FiveStar"
        stroke="none"
        stroke-width="1"
      >
        <g>
          <g>
            <rect
              fill="#00B67A"
              fill-rule="nonzero"
              height="96"
              id="Rectangle-path1"
              width="96"
              x="0"
              y="0"
            />
            <rect
              fill="#00B67A"
              fill-rule="nonzero"
              height="96"
              id="Rectangle-path2"
              width="96"
              x="104"
              y="0"
            />
            <rect
              fill="#00B67A"
              fill-rule="nonzero"
              height="96"
              id="Rectangle-path3"
              width="96"
              x="208"
              y="0"
            />
            <rect
              fill="#00B67A"
              fill-rule="nonzero"
              height="96"
              id="Rectangle-path4"
              width="96"
              x="312"
              y="0"
            />
            <rect
              fill="#00B67A"
              fill-rule="nonzero"
              height="96"
              id="Rectangle-path5"
              width="96"
              x="416"
              y="0"
            />
            <path
              d="M48,64.7 L62.6,61 L68.7,79.8 L48,64.7 Z M81.6,40.4 L55.9,40.4 L48,16.2 L40.1,40.4 L14.4,40.4 L35.2,55.4 L27.3,79.6 L48.1,64.6 L60.9,55.4 L81.6,40.4 L81.6,40.4 L81.6,40.4 L81.6,40.4 Z"
              fill="#FFFFFF"
              fill-rule="nonzero"
              id="Shape1"
            />
            <path
              d="M152,64.7 L166.6,61 L172.7,79.8 L152,64.7 Z M185.6,40.4 L159.9,40.4 L152,16.2 L144.1,40.4 L118.4,40.4 L139.2,55.4 L131.3,79.6 L152.1,64.6 L164.9,55.4 L185.6,40.4 L185.6,40.4 L185.6,40.4 L185.6,40.4 Z"
              fill="#FFFFFF"
              fill-rule="nonzero"
              id="Shape2"
            />
            <path
              d="M256,64.7 L270.6,61 L276.7,79.8 L256,64.7 Z M289.6,40.4 L263.9,40.4 L256,16.2 L248.1,40.4 L222.4,40.4 L243.2,55.4 L235.3,79.6 L256.1,64.6 L268.9,55.4 L289.6,40.4 L289.6,40.4 L289.6,40.4 L289.6,40.4 Z"
              fill="#FFFFFF"
              fill-rule="nonzero"
              id="Shape3"
            />
            <path
              d="M360,64.7 L374.6,61 L380.7,79.8 L360,64.7 Z M393.6,40.4 L367.9,40.4 L360,16.2 L352.1,40.4 L326.4,40.4 L347.2,55.4 L339.3,79.6 L360.1,64.6 L372.9,55.4 L393.6,40.4 L393.6,40.4 L393.6,40.4 L393.6,40.4 Z"
              fill="#FFFFFF"
              fill-rule="nonzero"
              id="Shape4"
            />
            <path
              d="M464,64.7 L478.6,61 L484.7,79.8 L464,64.7 Z M497.6,40.4 L471.9,40.4 L464,16.2 L456.1,40.4 L430.4,40.4 L451.2,55.4 L443.3,79.6 L464.1,64.6 L476.9,55.4 L497.6,40.4 L497.6,40.4 L497.6,40.4 L497.6,40.4 Z"
              fill="#FFFFFF"
              fill-rule="nonzero"
              id="Shape5"
            />
          </g>
        </g>
      </g>
    </svg>
  </div>
</div>
`;
